lib/rubycritic/source_control_systems/git.rb

```diff
@@ -52,6 +52,18 @@ def stashes_count
       def travel_to_original_state
         `git stash pop`
       end
+
+      def path_revisions_counts
+        `git log --name-only --format='' | sort | uniq -c`
```

Comment:
Thanks for the heads up!

1. Seems fixable if we process the output of the `git` command in ruby, rather than using `sort` and `uniq`. But I think the second issue will negate this one.
1. Ah, I didn't think about that! Seems possible to get around as well—though slightly more work. We can grab the name of the file along with what kind of change it was:
    ```shell
    $ git log --all --format='' --name-status --reverse
    ...
    A rubycritic.gemspec
    M rubycritic.gemspec
    M Rakefile
    M lib/rubycritic/report_generators/file_generator.rb
    M lib/rubycritic/report_generators/line_generator.rb
    R081  lib/rubycritic/report_generators/report_generator.rb  lib/rubycritic/report_generators/reporter.rb
    M lib/rubycritic/report_generators/base_generator.rb
    A lib/rubycritic/report_generators/index_generator.rb
    M lib/rubycritic/report_generators/reporter.rb
    ...
    ```
    Then parse the lines and build up the cache of changes:
    * A - add the path to the hash with value 1
    * B - ignore
    * C - add the copied path to the hash with value 1
    * D - remove the path from the hash
    * M - increment path value by 1
    * R - rename the path in the hash and increment the value by 1
    * T - increment path value by 1
    * U - ignore
    * X - ignore

    Since we log in reverse order, it should handle the case where a file named `foo` was renamed to `bar`, and later a new `foo` is introduced.

    Does that make sense?